File: sdks/typescript/src/resources/staff.ts
import { AxiosInstance } from 'axios';
import { PaginatedResponse } from '../types';

export interface Staff {
    id: string;
    user: string;
    role: string;
    branch?: string;
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

export class StaffMembers {
    constructor(private client: AxiosInstance) { }

    async list(page: number = 1, pageSize: number = 20, role?: string): Promise<PaginatedResponse<Staff>> {
        const params: any = { page, page_size: pageSize };
        if (role) params.role = role;
        const response = await this.client.get('/staff/', { params });
        return response.data;
    }

    async get(staffId: string): Promise<Staff> {
        const response = await this.client.get(`/staff/${staffId}/`);
        return response.data;
    }

    async create(data: Partial<Staff>): Promise<Staff> {
        const response = await this.client.post('/staff/', data);
        return response.data;
    }

    async update(staffId: string, data: Partial<Staff>): Promise<Staff> {
        const response = await this.client.patch(`/staff/${staffId}/`, data);
        return response.data;
    }

    async delete(staffId: string): Promise<void> {
        await this.client.delete(`/staff/${staffId}/`);
    }
}

File: sdks/rust/src/resources/staff.rs
use crate::models::{PaginatedResponse, Staff};
use crate::Puxbay;
use crate::PuxbayError;
use reqwest::Method;

pub struct StaffResource<'a> {
    pub(crate) client: &'a Puxbay,
}

impl<'a> StaffResource<'a> {
    pub fn new(client: &'a Puxbay) -> Self {
        Self { client }
    }

    pub async fn list(&self, page: i32, role: Option<&str>) -> Result<PaginatedResponse<Staff>, PuxbayError> {
        let mut query = format!("staff/?page={}", page);
        if let Some(r) = role {
            query.push_str(&format!("&role={}", r));
        }

        self.client.request(
            Method::GET,
            &query,
            None::<()>,
        ).await
    }

    pub async fn get(&self, staff_id: &str) -> Result<Staff, PuxbayError> {
        self.client.request(
            Method::GET,
            &format!("staff/{}/", staff_id),
            None::<()>,
        ).await
    }

    pub async fn create(&self, staff: &Staff) -> Result<Staff, PuxbayError> {
        self.client.request(
            Method::POST,
            "staff/",
            Some(staff),
        ).await
    }

    pub async fn update(&self, staff_id: &str, staff: &Staff) -> Result<Staff, PuxbayError> {
        self.client.request(
            Method::PATCH,
            &format!("staff/{}/", staff_id),
            Some(staff),
        ).await
    }

    pub async fn delete(&self, staff_id: &str) -> Result<(), PuxbayError> {
        self.client.request::<serde_json::Value>(
            Method::DELETE,
            &format!("staff/{}/", staff_id),
            None::<()>,
        ).await?;
        Ok(())
    }
}
